app/page/home/activity.tsx

```diff
@@ -51,182 +53,184 @@ const UserOpHistoryItem: React.FC<{
-          </div>
-        </div>
-      </a>
-    )
-  }
+  // Others' case
+  return (
+    <div className="w-full flex items-center">
+      {/* Status and to address */}
+      <Log calldata={callData} txStatus={txLog.status} />
+      {/* Transaction time */}
+      <TransactionTime time={creationTime} date={creationDate} />
+    </div>
+  )
+}
+
+const Log: React.FC<{
+  calldata: HexString
+  txStatus: TransactionStatus
+}> = ({ calldata, txStatus }) => {
+  const { chainId, tokens, type } = useAccount()
 
-  if (txLog.status === TransactionStatus.Rejected) {
+  const { to, value: nativeValue, data } = decodeExecuteParams(type, calldata)
+
+  let tokenName = `${getChainName(chainId).slice(0, 3)}ETH`
+  let value = nativeValue
+  let toAddress = to
+
+  // Send ETH
+  if (data === "0x") {
     return (
-      <div className="w-full flex items-center p-[14px_0px_14px_0px]">
-        <div className="flex-grow flex flex-col items-start">
-          {/* Status and image */}
-          <div className="flex items-center">
-            <OctagonXmark className="w-[16px] h-[16px] mr-[8px]" />
-            <div className="mr-[8px] leading-[19px] text-[16px] text-[#000000]">
-              {status}
-            </div>
-          </div>
-        </div>
-        {/* Transaction time */}
-        <div>
-          <div className="flex flex-col items-end">
-            <div className="mb-[4px] leading-[19px] text-[16px] text-[#000000]">
-              {creationDate}
-            </div>
-            <div className="leading-[19px] text-[16px] text-[#000000]">
-              {creationTime}
-            </div>
-          </div>
-        </div>
-      </div>
+      <TokenLog tokenName={tokenName} value={value} toAddress={toAddress}>
+        <StatusMark txStatus={txStatus} />
+      </TokenLog>
     )
   }
 
-  if (txLog.status === TransactionStatus.Sent) {
-    return (
-      <div className="w-full flex items-center p-[14px_0px_14px_0px]">
-        {/* Status and to address */}
-        <div className="flex-grow flex flex-col items-start">
-          {/* Status and token amount */}
-          <div className="flex items-center mb-[4px]">
-            <Clock className="w-[16px] h-[16px] mr-[8px]" />
-            <div className="mr-[8px] leading-[19px] text-[16px] text-[#000000]">
-              {status}
-            </div>
-            <div className="mr-[8px] leading-[19px] text-[16px] text-[#000000]">
-              1.2
-            </div>
-            <div className="leading-[19px] text-[16px] text-[#000000]">ETH</div>
+  // Interact with Dapp
+  const inTokenList = tokens.some((token) => {
+    if (address.isEqual(token.address, to)) {
+      try {
+        const { to, value: tokenValue } = decodeTransferParams(data)
+
+        tokenName = token.symbol
+        value = tokenValue
+        toAddress = to
+        return true
+      } catch (e) {
+        throw new Error(`Cannot decode token transfer: ${e}`)
+      }
+    }
+  })
+
+  return inTokenList ? (
```

Comment:
How about we rename `inTokenList` to be `isTokenTransfer` and set `isTokenTransfer = false` in the beginning?

Then, we could determine if the op is token transferred(ETH or ERC20) or interact with contract.
```
if (data === "0x") {
    isTokenTransfer = true
  } else {
    isTokenTransfer = tokens.some((token) => {
      if (address.isEqual(token.address, to)) {
        try {
          const { to, value: tokenValue } = decodeTransferParams(data)
          toAddress = to
          tokenName = token.symbol
          value = tokenValue
          return true
        } catch (e) {
          throw new Error(`Cannot decode token transfer: ${e}`)
        }
      }
    })
  }
```

Reply:
And I think we should return false instead of throwing in `catch` block.

Reply:
How about this change?

```typescript
  const isTokenTransfer =
    data === "0x" ||
    tokens.some((token) => {
      if (address.isEqual(token.address, to)) {
        try {
          const { to, value: tokenValue } = decodeTransferParams(data)

          tokenName = token.symbol
          value = tokenValue
          toAddress = to
          return true
        } catch (e) {
          return false
        }
      }
    })
```